app/jobs/sync_unattached_promo_registrations_stats_job.rb

```diff
@@ -0,0 +1,14 @@
+# Fetches and saves the referral stats for unattached codes
+class SyncUnattachedPromoRegistrationsStatsJob < ApplicationJob
+  include PromosHelper
+  queue_as :low
+
+  def perform(promo_id: active_promo_id)
+    promo_registrations = PromoRegistration.unattached_only
+    success = PromoRegistrationsStatsFetcher.new(promo_registrations: promo_registrations).perform
+
+    if success
+      Rails.cache.write("unattached_promo_registration_stats_last_synced_at", Time.now)
```

Comment:
Curious about where we fetch this, and if we're using this

Reply:
We fetch this here https://github.com/brave-intl/publishers/blob/1e80e15b001374ab8e7b2697c68bebce182ca095/app/views/admin/unattached_promo_registrations/index.html.slim#L27

That view is only accessible by admins.  I do have plans to display when the stats were last refreshed for publishers in the second part to resolve #1337